### ISSUE TITLE
*org-roam* buffer does not show Backlinks and Reflinks

### ISSUE THREAD

**Issue by bsiogs** [org-roam/org-roam#1708]
### Description
*org-roam* buffer does not show Backlinks and Reflinks with an error message: 
`No entry for org-roam-node-section in magit--section-type-alist`
#### Steps to Reproduce

1. Load Emacs
2. Open or create an org-roam file/node `foo`
3. Insert a link to an existing node `bar` (thanks to `C-r n i`)
4. Save  file containing `foo` node and close it
5. Open `bar` node
6. Call `*org-roam*` buffer
7. Links are empty and an error is displayed in mini-buffer `No entry for org-roam-node-section in magit--section-type-alist`
`org-roam` buffer configuration:
`(add-to-list 'display-buffer-alist 
             '("\\*org-roam\\*"
               (display-buffer-in-side-window)
               (side . right)
               (slot . 0)
               (window-width . 0.33)
               (window-parameters . ((no-other-window . t)
                                     (no-delete-other-windows . t)))))`

#### Backtrace
`Debugger entered--Lisp error: (error "BUG: No entry for org-roam-node-section in magit--...")
  signal(error ("BUG: No entry for org-roam-node-section in magit--..."))
  error("BUG: No entry for %s in %s" org-roam-node-section magit--section-type-alist)
  org-roam-node-insert-section(:source-node #s(org-roam-node :file "/.../.../.../orgRoam/dailies/2021-04-30.org" :file-hash nil :file-atime (24824 9698 773934 421000) :file-mtime (24821 42775 81243 958000) :id "49543dd1-8449-4650-aa82-37d72bc89434" :level 0 :point 1 :todo nil :priority nil :scheduled nil :deadline nil :title "2021-04-30" :properties (("CATEGORY" . "2021-04-30") ("ID" . "49543dd1-8449-4650-aa82-37d72bc89434") ("BLOCKED" . "") ("FILE" . "/.../.../.../orgRoam/dailies/2021-04-30.org") ("PRIORITY" . "B")) :olp nil :tags ("TDM" "optical_tweezers" "specimen_rotation" "JB") :aliases nil :refs nil) :point 164 :properties (:outline nil))
  org-roam-backlinks-section(#s(org-roam-node :file "/.../.../.../orgRoam/20200516112526-tdm.org" :file-hash nil :file-atime (24824 9698 769934 306000) :file-mtime (24821 42753 280914 466000) :id "b3855f2d-2a33-4539-8b37-8ef29a93806c" :level 0 :point 1 :todo nil :priority nil :scheduled nil :deadline nil :title "TDM" :properties (("CATEGORY" . "20200516112526-tdm") ("ID" . "b3855f2d-2a33-4539-8b37-8ef29a93806c") ("BLOCKED" . "") ("FILE" . "/.../.../.../orgRoam/20200516112526-tdm.org") ("PRIORITY" . "B")) :olp nil :tags nil :aliases nil :refs nil))
  org-roam-buffer-persistent-redisplay()
  org-roam-buffer-toggle()
  funcall-interactively(org-roam-buffer-toggle)
  call-interactively(org-roam-buffer-toggle nil nil)
  command-execute(org-roam-buffer-toggle)`

### Environment
- Emacs: GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 2.24.32)
 of 2021-07-19
- Framework: I don't know
- Org: Org mode version 9.4.4 (release_9.4.4 @ mixed installation! /.../.../.emacs.d/elpa/ox-tufte-latex-20160224.407/ and /usr/local/share/emacs/27.2/lisp/org/)
- Org-roam: 2.0.0 installed from Elpa `org-roam-20210719.930`

**Comment by carwin** [comment]
What is the value of your `org-roam-mode-section-functions` variable?

**Comment by bsiogs** [comment]
I ran
`(describe-variable 'org-roam-mode-section-functions)`
Here is the output :
`org-roam-mode-section-functions is a variable defined in ‘org-roam-mode.el’.
Its value is (org-roam-backlinks-section org-roam-reflinks-section) ...`

**Comment by nobiot** [comment]
How about upgrading your org-roam? Looks to be more than 10 days old; similar error reported and solved by upgrading: https://github.com/org-roam/org-roam/issues/1660

**Comment by bsiogs** [comment]
I downloaded last elpa accessible version `org-roam-20210730.916` and the issue is still here.

**Comment by nobiot** [comment]
I can’t reproduce the issue; I just upgraded to the same version via MELPA.

how do you do this, exactly?
- Call *org-roam* buffer

Are both nodes foo and bar file nodes?

**Comment by jethrokuan** [comment]
Remove any deferred loading for now, or add a `(require 'org-roam)`.

**Comment by emacs18** [comment]
I also ran into similar error regarding `magit-section` when I first migrated from v1 to v2.  I don't remember exactly how I resolved this.  My hunch is that re-installing packages such as `org-roam`, `magit-section`, `dash`, etc. may help.  I think I just re-installed all packages as I recall.  

Also I suggest `M-x org-roam-db-clear-all` followed by `M-x org-roam-db-sync` after making sure that `org-roam-directory` is set correctly.

Not sure if this is related, but magit related packages seem to be in transition to using melpa style version numbers, e.g., https://github.com/magit/magit/commit/ca4e65831c8111760ca209a87dca81ab233b5a38, which also impacted `magit-section` package.  Could version related problems have prevented proper updating of packages?
